easycv/datasets/detection/raw.py

```diff
@@ -31,9 +31,13 @@ def __len__(self):
         return self.num_samples
 
     def __getitem__(self, idx):
-        data_dict = self.data_source.get_sample(idx)
-        data_dict = self.pipeline(data_dict)
-        return data_dict
+        while True:
+            data_dict = self.data_source.get_sample(idx)
+            data_dict = self.pipeline(data_dict)
+            if data_dict is None:
+                idx = self.data_source._rand_another(idx)
```

Comment:
detection data source has no attribute  of `_rand_another`

Reply:
already modified